src/ast.js

```diff
@@ -3,24 +3,50 @@ const {symbolMap} = require('./constants');
 
 function traversSourceCode(source, parse) {
     const root = AST.parse(source, parse);
+    const sourceLines = source.split('\n');
     const nodes = [];
     let stack = [root];
 
+    function getSrc({start, end}) {
+        // Retrieve the source of the node
+        let src = null;
```

Comment:
You can use `Array.prototype.slice`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice
```
startLine = 1;
endLIne = 4;
arr = ['A', 'B', 'C', 'D', 'E', 'F'];
arr =arr.slice(startLine, endLIne).join('\n');
//output:
//B
//C
//D
```

Reply:
These aren't lines, they're coordinates objects, each containing line and column:
```
loc: {
    start: {line: 1, column:15},
    end: {line: 4, column: 1}
}
```